lib/rom/memory/dataset.rb

```diff
@@ -46,9 +46,24 @@ def project(*names)
 
       # Sort a dataset
       #
+      # @param [Array<Symbol>] names
+      #   Names of fields to order tuples by
+      #
+      # @option [Boolean] :nils_first (false)
+      #   Whether `nil` values should be placed before others
+      #
       # @api public
       def order(*names)
-        sort_by { |tuple| tuple.values_at(*names) }
+        options = names.last.is_a?(Hash) ? names.pop : {}
+        place   = options[:nils_first] ? -1 : 1
+        compare = ->(a, b) {
+          return a <=> b unless a.nil? ^ b.nil?
```

Comment:
Does the exclusive or here mean nils will be ordered using the normal natural order sort when only one value is nil? I think the partition and append/prepend method would be much cleaner?